### PR TITLE
Don't hardcode the sort order of the playlists.

### PR DESCRIPTION
**Pull request by JasonLG1979**
Currently the sort order in _getPlaylists is hardcoded to "Alphabetical".

This fixes that by checking the Orderings prop and using "Alphabetical" if "Alphabetical" is in the provided Orderings or if no Orderings are provided. Otherwise it uses the first sort order in the Orderings array if one is provided but "Alphabetical" is not in it.

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -208,6 +209,18 @@ const MPRISPlayer = new Lang.Class({
           if (props.ActivePlaylist)
             newState.playlist = props.ActivePlaylist.deep_unpack()[1][0];
 
+          if (props.Orderings) {
+            let orderings = ['Alphabetical'];
+            let maybeOrderings = props.Orderings.deep_unpack();
+            if (Array.isArray(maybeOrderings) && maybeOrderings.length > 0)
+              orderings = maybeOrderings;
```
Comment (eonpatapon) [inline src/player.js]: Maybe add a helper function for this code like:

```
function getOrderings(maybeOrderings) {
  let orderings = ['Alphabetical'];
  if (Array.isArray(maybeOrderings) && maybeOrderings.length > 0)
    orderings = maybeOrderings;
  return orderings;
}
```

File: src/player.js [hunk 2]
```diff
@@ -289,6 +302,14 @@ const MPRISPlayer = new Lang.Class({
...
       }
 
+      let orderings = this._mediaServerPlaylists.Orderings;
+      if (Array.isArray(orderings) && orderings.length > 0) {
+        newState.orderings = orderings;
+      }
+      else {
+        newState.orderings = ['Alphabetical'];
+      }
+
```
Comment (eonpatapon) [inline src/player.js]: So here you can also use it:

newState.orderings = getOrderings(this._mediaServerPlaylists.Orderings);

File: src/player.js [hunk 3]
```diff
@@ -457,6 +478,23 @@ const MPRISPlayer = new Lang.Class({
...
                           );
+      this._prop.GetRemote('org.mpris.MediaPlayer2.Playlists', 'Orderings',
+                           Lang.bind(this, function(value, err) {
+                             let state = new PlayerState();
+                             // default to ["Alphabetical"] if all else fails
+                             let orderings = ["Alphabetical"];
+                             if (!err) {
+                               let maybeOrderings = value[0].deep_unpack();
+                               if (Array.isArray(maybeOrderings) && maybeOrderings.length > 0)
+                                 orderings = maybeOrderings;
```
Comment (eonpatapon) [inline src/player.js]: And here also

**Comment by JasonLG1979** [comment]
@eonpatapon something like that?

**Comment by eonpatapon** [comment]
Looks good to me, thanks

**Pull request merged**